opm/simulators/linalg/bda/rocsparseSolverBackend.cpp

```diff
@@ -48,29 +48,41 @@
 #undef HIP_HAVE_CUDA_DEFINED
 #endif
 
-#define HIP_CHECK(stat)                               \
-    {                                                 \
-        if(stat != hipSuccess)                        \
-        {                                             \
-            OPM_THROW(std::logic_error, "HIP error"); \
-        }                                             \
-    }
-
-#define ROCSPARSE_CHECK(stat)                               \
-    {                                                       \
-        if(stat != rocsparse_status_success)                \
-        {                                                   \
-            OPM_THROW(std::logic_error, "rocsparse error"); \
-        }                                                   \
-    }
-
-#define ROCBLAS_CHECK(stat)                               \
-    {                                                     \
-        if(stat != rocblas_status_success)                \
-        {                                                 \
-            OPM_THROW(std::logic_error, "rocblas error"); \
-        }                                                 \
-    }
+#define HIP_CHECK(STAT)                                  \
+    do {                                                 \
+        const hipError_t stat = (STAT);                  \
+        if(stat != hipSuccess)                           \
+        {                                                \
+            std::ostringstream oss;                      \
+            oss << "rocsparseSolverBackend::hip ";       \
+            oss << "error: " << hipGetErrorString(stat); \
+            OPM_THROW(std::logic_error, oss.str());      \
+        }                                                \
+    } while(0)
```

Comment:
What is the reason for `do{...}while(0)` here? The condition is false, so it has no effect, right?

Reply:
standard macro idiom, see  e.g https://bruceblinn.com/linuxinfo/DoWhile.html for explanation.

Reply:
thanks.